driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java

```diff
@@ -287,37 +308,48 @@ public void call(final AsyncConnection connection, final Throwable t) {
         }
     }
 
-    static void validateReadConcernAndCollation(final Connection connection, final ReadConcern readConcern,
-                                                final Collation collation) {
+    static void validateFindOptions(final Connection connection, final ReadConcern readConcern, final Collation collation,
```

Comment:
Adding `allowDiskUse` to this method consolidates the options needing validation, so I changed the name of the method.

Reply:
Its slightly confusing as we have a `FindOptions` class but I can't think of a better alternative.

Reply:
I only use this method for `FindOperation` and reverted other operations to use `validateReadConcernAndCollation`.